### PR TITLE
 COD-4 Java application for function extraction [ci dev]

### PR DESCRIPTION
**Pull request by zohaib194**

### CODE REVIEW

File: backend/parser/src/main/java/me/codvis3D/ast/Main.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+                case "--File":
+                case "-f":
+                    file = argv[i + 1];
+                    i++;
+                    break;
+
+                default:
+                    System.err.println("[ERROR] Usage: java Main [-t | --Target] targetName [ -f | --File] fileName.");
+                    err = true;
```
Comment (Xillez) [inline backend/parser/src/main/java/me/codvis3D/ast/Main.java]: You should be able to only say return here and the function should quit.

File: backend/parser/src/main/java/me/codvis3D/ast/Main.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public static boolean err = false;
+    
+    public static void main(String[] argv) throws IOException {
+
+        parseArgs(argv);
+
+
+        // Checks which language to be parse.
+        switch(target){
+            case "cpp":
```
Comment (eldarht) [inline backend/parser/src/main/java/me/codvis3D/ast/Main.java]: use to upper to make all "CPP" same case or only allow lower case.

File: backend/parser/src/test/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import org.junit.jupiter.api.Test;
+
```
Comment (eldarht) [inline backend/parser/src/test/MainTest.java]: Not a valid test.

File: backend/parser/src/main/java/me/codvis3D/ast/Main.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    public static void parseArgs(String[] argv){
+        for (int i = 0; i < argv.length; i++) {
+            
+            flag = argv[i];
+
+            switch(flag){
+                case "--Target":
+                case "-t":
+                    target = argv[i + 1];
+                    i++;
```
Comment (Xillez) [inline backend/parser/src/main/java/me/codvis3D/ast/Main.java]: Could do this in the array accessing: 
```suggestion
                    target = argv[i++];
```
You're increasing it anyway just after

File: backend/parser/src/main/java/me/codvis3D/ast/Main.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    public static void parseArgs(String[] argv){
+        for (int i = 0; i < argv.length; i++) {
+            
+            flag = argv[i];
+
+            switch(flag){
+                case "--Target":
+                case "-t":
+                    target = argv[i + 1];
```
Comment (Xillez) [inline backend/parser/src/main/java/me/codvis3D/ast/Main.java]: Could do this for the array accessing: 
```suggestion
                    target = argv[i++];
```
You're increasing i just after anyway

Comment (Xillez) [inline backend/parser/src/main/java/me/codvis3D/ast/Main.java]: Could do this for the array accessing: 
```suggestion
                    target = argv[i++];
```
You're increasing i just after anyway

**Pull request merged**